test/default-xrp-client-test.ts

```diff
@@ -461,4 +469,130 @@ describe('Default Xpring Client', function(): void {
+            throw new Error(XRPClientErrorMessages.paymentConversionFailure)
+          } else {
+            expectedPaymentHistory.push(xrpTransaction)
+          }
+          break
+        }
+        default:
+      }
+    } // end for
+    // THEN the payment history is returned as expected
+    assert.deepEqual(expectedPaymentHistory, paymentHistory)
+  })
+
+  it('Payment History - classic address', function(done): void {
+    // GIVEN an XRPClient and a classic address
+    const xrpClient = new DefaultXRPClient(fakeSucceedingNetworkClient)
+    const classicAddress = 'rsegqrgSP8XmhCYwL9enkZ9BNDNawfPZnn'
+
+    // WHEN the payment history for an account is requested THEN an error to use X-Addresses is thrown.
+    xrpClient.paymentHistory(classicAddress).catch((error) => {
+      assert.typeOf(error, 'Error')
+      assert.equal(error.message, XRPClientErrorMessages.xAddressRequired)
+      done()
+    })
+  })
+
+  it('Payment History - network failure', function(done): void {
+    // GIVEN an XRPClient which wraps an erroring network client.
+    const xrpClient = new DefaultXRPClient(fakeErroringNetworkClient)
+
+    // WHEN the payment history is requested THEN an error is propagated.
+    xrpClient.paymentHistory(testAddress).catch((error) => {
+      assert.typeOf(error, 'Error')
+      assert.equal(error, FakeNetworkClientResponses.defaultError)
+      done()
+    })
+  })
+
+  it('Payment History - non-payment transactions', async function(): Promise<
+    void
+  > {
+    // GIVEN an XRPClient client which will return a transaction history which contains non-payment transactions
+
+    // Generate expected transactions from the default response, which only contains payments.
+    const nonPaymentTransactionResponse = new GetTransactionResponse()
+    nonPaymentTransactionResponse.setTransaction(testCheckCashTransaction)
+    const heteregeneousTransactionHistory = testGetAccountTransactionHistoryResponse // here, homogeneous w/ PAYMENT txns
+    heteregeneousTransactionHistory.addTransactions(
+      // add a CHECKCASH txn type
+      nonPaymentTransactionResponse,
+    )
+
+    const heteroHistoryNetworkResponses = new FakeNetworkClientResponses(
+      FakeNetworkClientResponses.defaultAccountInfoResponse(),
+      FakeNetworkClientResponses.defaultFeeResponse(),
+      FakeNetworkClientResponses.defaultSubmitTransactionResponse(),
+      FakeNetworkClientResponses.defaultGetTransactionResponse(),
+      heteregeneousTransactionHistory,
+    )
+
+    const heteroHistoryNetworkClient = new FakeNetworkClient(
+      heteroHistoryNetworkResponses,
+    )
+
+    const xrpClient = new DefaultXRPClient(heteroHistoryNetworkClient)
+
+    // WHEN the transactionHistory is requested.
+    const transactionHistory = await xrpClient.paymentHistory(testAddress)
+
+    // THEN the returned transactions are conversions of the inputs with non-payment transactions filtered.
+    const expectedTransactionHistory = [
```

Comment:
Your input was `testGetAccountTransactionHistoryResponse` so I think a small improvement here could be to loop through that and convert the payments.

Reply:
So I did update this... but that logic for looping through the transactionHistory proto and converting it into an array of XRPTransactions happens in several places here (the `paymentHistory` implementation, and two of the test cases for default-xrp-client), which feels bad.   If I were to factor this into its own method, where would you think that method should belong?  Or should I not, since we don't care about repeating ourselves as much if it's just test code?

Reply:
So an alternative here which is really simple is that the constructor for `XRPTransaction` will return undefined if the given input is not a payment. We use a switch in the actual library because we want to catch the case where:
- The transaction is a payment
- Conversion fails

And and throw an error rather than returning an incomplete set of data (which someone may then use to make incorrect choices based on incomplete data). 

You could really write:
```javascript
var xrpTxs = []
for (var i = 0; i < transactions.length; i++) {
  const tx = transactions[i]
  const xrpTX = XRPTransaction.from(tx)
  if (!xrpTx) {
      continue
  }
  xtpTxs.push(xrpTx)
}
```

Your way also works. 

With regards to where to put this code, I wouldn't worry about deduping test code with the actual code base (there's no obvious place to put this logic without making it public which feels like an over optimization). If you're repeating yourself in the tests then you could just write a helper in  this file, or if you need to share it across multiple files you could make a `XRPTestUtils` class in `test/XRP/Helpers`